### PR TITLE
ILOS (I Lose ollmy Sanity)

### PR DESCRIPTION
**Pull request by karla-vm**
### Description

Because of how we handle the form templates in MCR, hiding a route behind a feature flag on the frontend doesn't actually cover any completion status problems, so this PR handles the ILOS requirement a bit more cleanly:

If the ILOS feature flag is enabled, the form template will include ILOS fields. If that feature flag is disabled, then it won't.

### Related ticket(s)

N/A

---
### How to test

In this deployed instance or locally, you should still see the ILOS fields for MCPAR. If you hardcode `ilosAvailable` to `false` [here](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/blob/4ae3940d7479676c87994de1e85cd75c05d27a39/services/ui-src/src/components/modals/AddEditReportModal.tsx#L96) those sections won't be there 🪄 

### Notes

N/A

---
### Pre-review checklist

- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [ ] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

---
#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: services/ui-src/src/utils/forms/ilosFields.ts [hunk 1]
```diff
@@ -73,3 +73,22 @@ const updatedIlosChoiceList = (
   });
   return updatedChoiceList;
 };
+
+// if reporting on ILOS, verify that there its questions are completed
```
Comment (gmrabian) [inline services/ui-src/src/utils/forms/ilosFields.ts:77]: typo in comment

File: services/app-api/utils/formTemplates/formTemplates.ts [hunk 2]
```diff
@@ -323,3 +327,21 @@ const makePCCMTemplateModifications = (reportTemplate: ReportJson) => {
...
+  reportTemplate.routes = reportTemplate.routes.map((route: ReportRoute) => {
+    if (
+      route.path === "/mcpar/program-information" ||
+      route.path === "/mcpar/plan-level-indicators"
+    ) {
+      const updatedChildren = route.children?.slice(0, -1);
+      route.children = updatedChildren;
+    }
+    return route;
+  });
```
Comment (benmartin-coforma) [inline services/app-api/utils/formTemplates/formTemplates.ts:344]: IDEA: I think I would use `for..of` here. We're already working within a structuredClone, so there's no harm in skipping the map to do these updates in place.

```suggestion
  for (let route of reportTemplate.routes) {
    if (route.path === "/mcpar/program-information" ||
      route.path === "/mcpar/plan-level-indicators"
    ) {
      // These sections' last subsection is ILOS-specific; remove it.
      route.children = route.children?.slice(0, -1);
    }
  }
```

**Review by gmrabian — DISMISSED** [review]

File: services/ui-src/src/components/modals/AddEditReportModal.tsx [hunk 3]
```diff
@@ -38,7 +38,10 @@ export const AddEditReportModal = ({
   const { copyEligibleReportsByState } = useStore();
 
   const [submitting, setSubmitting] = useState<boolean>(false);
+
+  // LaunchDarkly
   const yoyCopyFlag = useFlags()?.yoyCopy;
+  const ilos = useFlags()?.ilos;
```
[the remark this answers is not in the document]
Comment (karla-vm) [reply services/ui-src/src/components/modals/AddEditReportModal.tsx:44]: We define them in the LD portal as boolean, correct.

File: services/ui-src/src/components/modals/AddEditReportModal.tsx [hunk 4]
```diff
@@ -90,6 +93,7 @@ export const AddEditReportModal = ({
       formData["reportingPeriodEndDate"]
     );
     const programIsPCCM = formData["programIsPCCM"];
+    const ilosAvailable = ilos;
```
Comment (bangbay-bluetiger) [inline services/ui-src/src/components/modals/AddEditReportModal.tsx:96]: Is this `const` needed? Could assign to `ilos` to `ilosAvailable` directly in the metadata hash.

Comment (karla-vm) [reply services/ui-src/src/components/modals/AddEditReportModal.tsx]: good point! removed it.

**Review by gmrabian — APPROVED** [review]

**Review by gmrabian — APPROVED** [review]

**Review by jessabean — APPROVED** [review]

**Pull request merged**